tests/client/test_orion_client.py

```diff
@@ -56,6 +59,107 @@ def test_get_client_cache_uses_profile_settings(self):
             assert new_client is not client
 
 
+class TestClientProxyAwareness:
+    """Regression test for https://github.com/PrefectHQ/nebula/issues/2356, where
+    a customer reported that the Cloud client supported proxies, but the Orion client
+    did not.  This test suite is implementation-specific to httpx/httpcore, as there are
+    no other inexpensive ways to confirm both the proxy-awareness and preserving the
+    retry behavior without probing into the implementation details of the libraries."""
+
+    @pytest.fixture()
+    def remote_https_orion(self) -> Generator[httpx.URL, None, None]:
+        orion_url = "https://127.0.0.1:4242/"
+        with temporary_settings(updates={PREFECT_API_URL: orion_url}):
+            yield httpx.URL(orion_url)
+
+    def test_unproxied_remote_client_will_retry(self, remote_https_orion: httpx.URL):
+        """The original issue here was that we were overriding the `transport` in
+        order to set the retries to 3; this is what circumvented the proxy support.
+        This test (and those below) should confirm that we are setting the retries on
+        the transport's pool in all cases."""
+        httpx_client = get_client()._client
+        assert isinstance(httpx_client, httpx.AsyncClient)
+
+        transport_for_orion = httpx_client._transport_for_url(remote_https_orion)
+        assert isinstance(transport_for_orion, httpx.AsyncHTTPTransport)
+
+        pool = transport_for_orion._pool
+        assert isinstance(pool, httpcore.AsyncConnectionPool)
+        assert pool._retries == 3  # set in prefect.client.orion.get_client()
+
+    def test_users_can_still_provide_transport(self, remote_https_orion: httpx.URL):
```

Comment:
@madkinsz Here's the new test case for allowing the user to provide a `transport`, great catch!

Reply:
Thanks!